generative/networks/nets/patchgan_discriminator.py

```diff
@@ -67,31 +70,66 @@ def __init__(
     ) -> None:
         super().__init__()
         self.num_d = num_d
+        if isinstance(num_layers_d, int) and pooling_method is None:
+            # if pooling_method is None, calculate the number of layers for each discriminator by multiplying by the number of discriminators
+            num_layers_d = [num_layers_d * i for i in range(1, num_d + 1)]
+        elif isinstance(num_layers_d, int) and pooling_method is not None:
+            # if pooling_method is not None, the number of layers is the same for all discriminators
+            num_layers_d = [num_layers_d] * num_d
         self.num_layers_d = num_layers_d
+        assert (
+            len(self.num_layers_d) == self.num_d
+        ), f"MultiScalePatchDiscriminator: num_d {num_d} must match the number of num_layers_d. {num_layers_d}"
+
+        if pooling_method is None:
+            pool = None
+        else:
+            pool = get_pool_layer(
+                (pooling_method, {"kernel_size": kernel_size, "stride": 2}), spatial_dims=spatial_dims
+            )
```

Comment:
Apparently, the default padding here is 0, which is not a desirable behavior. Could you add "padding": (kernel_size-1)//2? This way it will be the same shape, divided by 2.